metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -133,8 +140,7 @@ class DebugProvider(
 
       compilations.compilationFinished(targets).flatMap { _ =>
         buildServer
-          .startDebugSession(parameters)
-          .withTimeout(60, TimeUnit.SECONDS)
+          .startDebugSession(parameters, cancelPromise)
```

Comment:
What about clients that aren't `slowTaskProviders`. If something is going wrong, should we still have a default timeout for them?

Reply:
Hmm... we could have it as a default, but that isn't too pretty there. I can think of two solutions:
1. show message request after timeout to see if they really want to cancel <- not everything might support it
2. add cancel token for executeCommand, which means we would make the editor responsible for cancelling there.

I could take a look further if maybe we could manage to use 2. from the editor side.

Reply:
Hmm... Vs Code doesn't have anything in the UI to cancel debug session before it starts

Reply:
So most likely it's either 1. or 3. meaning we leave the timeout for non slowTaskProvider

Reply:
> show message request after timeout to see if they really want to cancel <- not everything might support it

I think _maybe_ this would make sense, but yea it probably won't be pretty.